### PR TITLE
ISPN-12383 improve marshalling docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-12383

### CODE REVIEW

File: documentation/src/main/asciidoc/stories/assembly_marshalling_user_types.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+= Marshalling User Types with ProtoStream
+User types are Java objects that {brandname} does not support out of the box.
+To marshall user types, you implement the `SerializationContextInitializer`
+interface to describe your Java objects so that the ProtoStream library can
+encode them to Protobuf format and {brandname} can transmit and store them.
+
+include::{topics}/proc_generating_protostream_sci.adoc[leveloffset=+1]
+include::{topics}/proc_manually_registering_sci.adoc[leveloffset=+2]
+include::{topics}/proc_implementing_protostream_sci.adoc[leveloffset=+1]
+include::{topics}/proc_adding_deserialization_allowlist.adoc[leveloffset=+1]
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/stories/assembly_marshalling_user_types.adoc:12]: This isn't required for protostream, so this should just be linked from the docs of Marshallers that require it, e.g. `JavaSerializationMarshaller` or `JbossUserMarshaller`.

File: documentation/src/main/asciidoc/topics/proc_implementing_protostream_sci.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
++
+. Create a `SerializationContextInitializer` implementation that registers the
+`.proto` schema and the ProtoStream marshaller implementations with a
+`SerializationContext`.
++
+[source,java]
+.ManualSerializationContextInitializer.java
+----
+include::code_examples/ManualSerializationContextInitializer.java[]
+----
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_implementing_protostream_sci.adoc:56]: When manually creating a SCI it's necessary to register it with the configuration as described in `Manually Registering Serialization Context Initializers`. This is a usability issue until Adrian's new API is available, as it means that if users have to create a single manual SCI for external types, then they also have to register all generated SCIs manually.

**Review by ryanemerson — CHANGES_REQUESTED** [review]
Overall this is a big improvement :slightly_smiling_face:

File: documentation/src/main/asciidoc/stories/assembly_marshaller_implementations.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[id='marshaller_implementations']
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/stories/assembly_marshaller_implementations.adoc:1]: Is this story included anywhere?

File: documentation/src/main/asciidoc/stories/assembly_marshalling_user_types.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/stories/assembly_marshalling_user_types.adoc:12]: ah... OK. I thought it was required for all cases where user objects were used. will fix.

File: documentation/src/main/asciidoc/stories/assembly_marshaller_implementations.adoc (hunk 3 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/stories/assembly_marshaller_implementations.adoc:1]: yep. it's nested under the main `assembly_marshalling.adoc` story.

File: documentation/src/main/asciidoc/topics/proc_implementing_protostream_sci.adoc (hunk 2 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_implementing_protostream_sci.adoc:49]: thanks! fixed.

**Review by ryanemerson — APPROVED** [review]

**Pull request merged**